src/cartesi-machine.lua

```diff
@@ -2007,83 +1969,63 @@ local function load_memory_range(machine, config, filename)
-    local length = string.unpack(">I8", machine:read_memory(config.start + 3 * 32 - 8, 8))
-    -- add address, offset, and payload length to amount to be read
-    length = length + 3 * 32
-    assert(f:write(machine:read_memory(config.start, length)))
-    f:close()
-end
-
-local function save_rollup_advance_state_notice(machine, config, advance)
-    local values = { e = advance.epoch_index, i = advance.next_input_index - 1, o = advance.notice_index }
-    local name = instantiate_filename(advance.notice, values)
-    stderr("Storing %s\n", name)
-    local f = assert(io.open(name, "wb"))
-    -- skip offset to reach payload length
-    local length = string.unpack(">I8", machine:read_memory(config.start + 2 * 32 - 8, 8))
-    -- add offset and payload length to amount to be read
-    length = length + 2 * 32
-    assert(f:write(machine:read_memory(config.start, length)))
-    f:close()
+local function dump_exception(machine, config, length)
+    local payload = machine:read_memory(config.start, length)
+    stderr("cmio exception with payload: %q\n", payload)
 end
 
-local function dump_exception(machine, config)
-    -- skip offset to reach payload length
-    local length = string.unpack(">I8", machine:read_memory(config.start + 2 * 32 - 8, 8))
-    -- add offset and payload length to amount to be read
-    local payload = machine:read_memory(config.start + 2 * 32, length)
-    stderr("Rollup exception with payload: %q\n", payload)
-end
-
-local function save_rollup_advance_state_report(machine, config, advance)
-    local values = { e = advance.epoch_index, i = advance.next_input_index - 1, o = advance.report_index }
-    local name = instantiate_filename(advance.report, values)
-    stderr("Storing %s\n", name)
-    local f = assert(io.open(name, "wb"))
-    -- skip offset to reach payload length
-    local length = string.unpack(">I8", machine:read_memory(config.start + 2 * 32 - 8, 8))
-    -- add offset and payload length to amount to be read
-    length = length + 2 * 32
-    assert(f:write(machine:read_memory(config.start, length)))
-    f:close()
-end
-
-local function save_rollup_inspect_state_report(machine, config, inspect)
+local function save_cmio_inspect_state_report(machine, config, inspect, length)
     local values = { o = inspect.report_index }
     local name = instantiate_filename(inspect.report, values)
     stderr("Storing %s\n", name)
     local f = assert(io.open(name, "wb"))
-    -- skip offset to reach payload length
-    local length = string.unpack(">I8", machine:read_memory(config.start + 2 * 32 - 8, 8))
-    -- add offset and payload length to amount to be read
-    length = length + 2 * 32
     assert(f:write(machine:read_memory(config.start, length)))
     f:close()
 end
 
+local function check_outputs_root_hash(root_hash, output_hashes)
+    local z = string.rep("\0", 32)
+    if #output_hashes == 0 then
+        output_hashes = { z }
+    end
+    for i = 1, 64 do
```

Comment:
Suppress `cartesi-machine.lua:2004:9: unused loop variable i` warning with:

```suggestion
    for _ = 1, 64 do
```